third_party/move/move-model/src/model.rs

```diff
@@ -3244,7 +3309,7 @@ pub struct FunctionData {
 
     /// Optional definition associated with this function. The definition is available if
     /// the model is build with option `ModelBuilderOptions::compile_via_model`.
-    pub(crate) def: Option<Exp>,
+    pub(crate) def: RefCell<Option<Exp>>,
```

Comment:
Now I see why you need the borrow. However, I wonder: is this really needed? IIRC your inliner takes `&mut GlobalEnv` so could we not just use that for mutation?

Reply:
Additional note here: perhaps we should start having a `RefCell` here anyway, for example subsequent simplifier or AST based processing. This is much easier to do with `&GlobalEnv`.

Reply:
Unfortunately, this field is on `FunctionData`, and `FuncEnv` has a field `data: &'env FunctionData`.   We can't make that field `&mut` because then `FunctionEnv` can't satisfy `Clone`.  This is a big can of worms.